components/style/properties/properties.mako.rs

```diff
@@ -313,6 +313,36 @@ impl PropertyDeclarationBlock {
     }
 }
 
+impl PropertyDeclarationBlock {
+    pub fn to_css_single_value<W>(&self, dest: &mut W, name: &str)
+        -> fmt::Result where W: fmt::Write {
+        if self.declarations.len() == 0 {
+            Err(fmt::Error)
+        } else if self.declarations.len() == 1 &&
+                  self.declarations[0].0.name().to_string() == name {
```

Comment:
Oh, I see why I did this -- it was temporary. `PropertyDeclarationName` is an enum with no `PartialEq<&str>` implementation. I'll add something for case insensitive comparisons.